packages/dd-trace/test/appsec/sdk/user_blocking.spec.js

```diff
@@ -0,0 +1,233 @@
+'use strict'
+
+const proxyquire = require('proxyquire')
+const agent = require('../../plugins/agent')
+const tracer = require('../../../../../index')
+const appsec = require('../../../src/appsec')
+const Config = require('../../../src/config')
+const getPort = require('get-port')
+const axios = require('axios')
+const RuleManager = require('../../../src/appsec/rule_manager')
+const fs = require('fs')
+const crypto = require('crypto')
+
+describe('User blocking API', () => {
+  describe('Test internal API', () => {
+    const tracer = {}
+    const mockReq = { protocol: 'https' }
+    const mockRes = { headersSent: false }
+    const storage = { getStore: () => {
+      return { req: mockReq, res: mockRes }
+    } }
+
+    let block, getRootSpan, mockRootSpan, mockSetTag, userBlocking
+
+    beforeEach(() => {
+      block = sinon.stub()
+      mockSetTag = sinon.stub()
+      mockRootSpan = {
+        context: () => {
+          return { _tags: { 'usr.id': 'mockUser' } }
+        },
+        setTag: mockSetTag
+      }
+      getRootSpan = sinon.stub().returns(mockRootSpan)
+      userBlocking = proxyquire('../../../src/appsec/sdk/user_blocking', {
+        './utils': { getRootSpan },
+        '../blocking': { block },
+        '../../../../datadog-core': { storage }
+      })
+    })
+
+    it('checkUserAndSetUser should return false with an empty user', () => {
+      const user = {}
+      const ret = userBlocking.checkUserAndSetUser(tracer, user)
+      expect(ret).to.be.false
+    })
+
+    it('checkUserAndSetUser should return false with no user', () => {
+      const ret = userBlocking.checkUserAndSetUser()
+      expect(ret).to.be.false
+    })
+
+    it('blockRequest should call block with proper arguments', () => {
+      userBlocking.blockRequest(tracer, {}, {})
+      expect(block).to.be.calledOnceWithExactly({}, {}, mockRootSpan)
+    })
+
+    it('blockRequest should get req and res from local storage when they are not passed', () => {
+      userBlocking.blockRequest(tracer)
+      expect(block).to.be.calledOnceWithExactly(mockReq, mockRes, mockRootSpan)
+    })
+
+    it('blockRequest should return proper value when there is no rootSpan available', () => {
+      getRootSpan.returns(undefined)
+
+      const ret = userBlocking.blockRequest(tracer, {}, {})
+      expect(ret).to.be.false
+      expect(block).not.to.have.been.called
+    })
+
+    it('checkUserAndSetUser should return false when there is no rootSpan available', () => {
+      getRootSpan.returns(undefined)
+
+      const ret = userBlocking.checkUserAndSetUser(tracer, { id: 'user' })
+      expect(getRootSpan).to.be.calledOnceWithExactly(tracer)
+      expect(ret).to.be.false
+    })
```

Comment:
this test is totally wrong and only returns false because the gateway doesn't match anything! I'll fix it in a sec